docs/rfcs/0005-coupons.md

```diff
@@ -0,0 +1,102 @@
+is redeemable for is different from MicroMasters.
+
+A program coupon is redeemable only if all courses for the program are in the basket. If a course
+was missing from the basket, maybe because the program is nearing end of life and one of
+the courses is not enrollable anymore, then the program coupon would not be redeemable at all.
+  
+A course coupon would be redeemable only for the next course run for that course. It may have 
+multiple courses where it can be redeemed.
+
+A course run coupon is redeemable only for the particular course run.
+
+Note: The UI will not include a basket, the basket will only be in the backend. The end user
+will only be able to buy one course or one program at a time, and can apply only one coupon.
+This should simplify the UX for redeeming coupons.
+
+##### Number of redemptions
+
+Coupons have a limited number of total redemptions. They cannot be unlimited to ensure that
+there's a cap on the number of discounts applied.
+However this could be a very high number, higher than is likely to ever be redeemed.
+
+TODO: Does this sound right? If the number is actually unlimited we should handle it that way.
+
+A coupon will also have a limited number of redemptions per user. Most coupons will only allow
+one redemption per user. Automatic program discounts will have the number of redemptions per user
+be the same as the total number of redemptions.
+
+A coupon may be used by any number of users as long as there are redemptions left.
+
+##### Validity
+
+Coupons can be enabled or disabled via a field, or enabled only for a certain period of time.
+If a coupon was not enabled at the time of purchase it would not be redeemable for anything.
+
+A coupon may also become invalid if the related product is not available anymore.
+
+### Use cases
+
+#### One time use coupons
+
+We ask the admin page to generate 1000 coupons for a course and we email learners individually
+with the coupon code. It is a percent discount for one course.
+
+##### Named coupons
+
+A coupon is distributed widely to the general public with a special name,
+for example `MARCHMADNESS15` to get 15% off. The coupon would have a very high number of
+redemptions.
+
+The admin would select the courses which this coupon would apply to. The coupon would work
+with any of these courses, but the user would only be able to redeem the coupon for one of the courses.
+The coupon would not be redeemable for that user for the second time.
+
+TODO: any fear of users making separate accounts to redeem the coupon twice?
+
+##### Automatic program discount
+
+A user buys a program and gets 15% off per course. The coupon is automatically
+applied at checkout. They will not be able to apply a second coupon on top of this.
+
+The UX for this should work fine because we only plan to actually provide coupons for
+individual courses. So there would not be a coupon that would work for a program level discount,
+other than the one automatically applied.
```

Comment:
I'm not sure this is correct "we only plan to actually provide coupons for individual courses." Let me verify.

Reply:
Ok, I've confirmed the mitxpro currently offers coupons for some single courses (Data Science and Additive Manufacturing) and some programs (Systems Engineering). 

When there is a program like  Systems Engineering, they focus the marketing on getting people to purchase the program bundle.

Reply:
@pdpinch can you elaborate ?

Reply:
@pdpinch are those meant to be automatically applied discounts?

I'm thinking given that there is no use case [here](https://docs.google.com/document/d/1QEmmEfgmO2s2tJnbbXDaOMXVKWs8c5Y-lCiFre-vTdY/edit) for automatic discounts, that we should remove this use case and figure it out later on when it comes up.

Reply:
Sorry, I forgot this paragraph is in the context of automatic program discounts. 

The point I want to make is that while there will be an automatic program bundle discount (for example, 10% off on System Engineering), the marketing team may need to send out a named coupon that's is better than the automatic discount (for example, 20% off on System Engineering). 

In that scenario, when a user applies the 20% named coupon to their program, it would be a better discount and supersede the automatic program bundle discount.